centreon/tests/e2e/features/Dashboards/10-widget-metric-graph-configuration/index.ts

```diff
@@ -0,0 +1,398 @@
+  cy.getByTestId({ testId: 'Avg' }).eq(1).should('exist');
+});
+
+Then('the second widget has the same properties as the first widget', () => {
+  cy.getByTestId({ testId: 'critical-line-400-tooltip' }).eq(1).should('exist');
+  cy.getByTestId({ testId: 'warning-line-200-tooltip' }).eq(1).should('exist');
+});
+
+Given('a dashboard featuring two Metrics Graph widgets', () => {
+  cy.insertDashboardWithMetricsGraphWidget(
+    dashboards.default,
+    metricsGraphDoublecWidget
+  );
+  cy.visit(`${Cypress.config().baseUrl}/centreon/home/dashboards`);
+  cy.wait('@listAllDashboards');
+  cy.getByLabel({
+    label: 'view',
+    tag: 'button'
+  })
+    .contains(dashboards.default.name)
+    .click();
+  cy.getByLabel({
+    label: 'Edit dashboard',
+    tag: 'button'
+  }).click();
+  cy.getByTestId({ testId: 'More actions' }).eq(0).click();
+  cy.wait('@performanceData');
+});
+
+When(
+  'the dashboard administrator user deletes one of the Metrics Graph widgets',
+  () => {
+    cy.getByTestId({ testId: 'DeleteIcon' }).click();
+    cy.getByLabel({
+      label: 'Delete',
+      tag: 'li'
+    }).realClick();
+  }
+);
+
+Then(
+  'only the contents of the other Metrics Graph widget are displayed',
+  () => {
+    cy.getByTestId({ testId: 'Min' }).should('exist');
+    cy.getByTestId({ testId: 'Max' }).should('exist');
+    cy.getByTestId({ testId: 'Avg' }).should('exist');
+  }
+);
+
+Given('a dashboard featuring a configured Metrics Graph widget', () => {
+  cy.insertDashboardWithMetricsGraphWidget(
+    dashboards.default,
+    metricsGraphWidget
+  );
+  cy.visit(`${Cypress.config().baseUrl}/centreon/home/dashboards`);
+  cy.wait('@listAllDashboards');
+  cy.getByLabel({
+    label: 'view',
+    tag: 'button'
+  })
+    .contains(dashboards.default.name)
+    .click();
+  cy.getByLabel({
+    label: 'Edit dashboard',
+    tag: 'button'
+  }).click();
+  cy.getByTestId({ testId: 'MoreVertIcon' }).click();
+  cy.getByLabel({
+    label: 'Edit widget',
+    tag: 'li'
+  }).realClick();
+  cy.wait('@performanceData');
+});
+
+When(
+  'the dashboard administrator user selects a metric with a different unit than the initial metric in the dataset selection',
+  () => {
+    cy.getByTestId({ testId: 'Select metric' }).click();
+    cy.get('[class^="MuiAutocomplete-option"]').eq(1).realClick();
+  }
+);
+
+Then(
+  'additional bars representing the metric behavior of these metrics are added to the Metrics Graph widget',
+  () => {
+    cy.getByTestId({ testId: 'Min' }).eq(1).should('exist');
+    cy.getByTestId({ testId: 'Max' }).eq(1).should('exist');
+    cy.getByTestId({ testId: 'Avg' }).eq(1).should('exist');
+  }
+);
+
+Then(
+  'an additional Y-axis based on the unit of these additional bars is displayed',
+  () => {
+    cy.get('p[aria-label="Centreon-Server: Packet Loss (%)"]').should('exist');
+  }
+);
+
+Then('the thresholds are automatically hidden', () => {
+  cy.get('span[data-checked="false"]').should('exist');
```

Comment:
Regarding the step name, I expect `should('not.be.visible')`

Reply:
When the input switch is activated, it is represented by data-checked="true". Conversely, when it is not activated, the representation is data-checked="false". To ensure that the input switch is in the deactivated state, I have included the following command: cy.get('span[data-checked="false"]').should('exist');

Reply:
I understand your point, but can you add an assertion which check thresholds are not visible